src/libdrakvuf/win-processes.c

```diff
@@ -313,23 +313,36 @@ static unicode_string_t* win_get_process_full_name(drakvuf_t drakvuf, addr_t epr
 
 char* win_get_process_name(drakvuf_t drakvuf, addr_t eprocess_base, bool fullpath)
 {
-    if ( fullpath )
-    {
-        unicode_string_t* fullname = win_get_process_full_name( drakvuf, eprocess_base );
+    unicode_string_t* fullname = win_get_process_full_name( drakvuf, eprocess_base );
 
-        if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)
+    if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)
+    {
+        // Replace 'proc_data->name' with 'fullname->contents'
+        // Moving ownership of fullname->contents to name for later cleanup
+        char* name = (char*)fullname->contents;
+        g_free( (gpointer)fullname );
+        // ImageFileName size differs between kernel builds & versions,
+        // relying on it would sometimes yield incomplete process name.
+        if ( !fullpath )
         {
-            // Replace 'proc_data->name' with 'fullname->contents'
-            // Moving ownership of fullname->contents to name for later cleanup
-            char* name = (char*)fullname->contents;
-            g_free( (gpointer)fullname );
-            return name;
+            char** tokens = g_strsplit(name, "\\", -1);
+            int index = 0;
+
+            while (tokens && tokens[index+1] != NULL)
+                index++;
+
+            if (tokens)
+                name = g_strdup(tokens[index]);
```

Comment:
You can simplify all this to be just:
```
while(tokens[index])
    name=tokens[index++];
```

Reply:
@tklengyel  Indeed but it throws a null pointer deference in the CI

Reply:
I've tried something similar, but I believe this will give a CI error null pointer deference, let's see how it goes.